src/Handler/Workspace/Create/CreateWorkspaceHandler.php

```diff
@@ -0,0 +1,118 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\StorageDriver\Teradata\Handler\Workspace\Create;
+
+use Google\Protobuf\Internal\Message;
+use Keboola\StorageDriver\Command\Workspace\CreateWorkspaceCommand;
+use Keboola\StorageDriver\Command\Workspace\CreateWorkspaceResponse;
+use Keboola\StorageDriver\Contract\Driver\Command\DriverCommandHandlerInterface;
+use Keboola\StorageDriver\Credentials\GenericBackendCredentials;
+use Keboola\StorageDriver\Shared\BackendSupportsInterface;
+use Keboola\StorageDriver\Shared\NameGenerator\NameGeneratorFactory;
+use Keboola\StorageDriver\Shared\Utils\Password;
+use Keboola\StorageDriver\Teradata\TeradataSessionManager;
+use Keboola\TableBackendUtils\Escaping\Teradata\TeradataQuote;
+
+final class CreateWorkspaceHandler implements DriverCommandHandlerInterface
+{
+    public const DEFAULT_PERM_SPACE_SIZE = 1e8; // 100MB
+    public const DEFAULT_SPOOL_SPACE_SIZE = 1e8; // 100MB
+
+    private TeradataSessionManager $manager;
+
+    public function __construct(TeradataSessionManager $manager)
+    {
+        $this->manager = $manager;
+    }
+
+    /**
+     * @inheritDoc
+     * @param GenericBackendCredentials $credentials
+     * @param CreateWorkspaceCommand $command
+     */
+    public function __invoke(
+        Message $credentials,
+        Message $command,
+        array $features
+    ): ?Message {
+        assert($credentials instanceof GenericBackendCredentials);
+        assert($command instanceof CreateWorkspaceCommand);
+
+        // validate
+        assert(!empty($command->getProjectId()), 'CreateWorkspaceCommand.projectId is required');
+        assert(!empty($command->getWorkspaceId()), 'CreateWorkspaceCommand.workspaceId is required');
+        assert(!empty($command->getProjectUserName()), 'CreateWorkspaceCommand.projectUserName is required');
+        assert(!empty($command->getProjectRoleName()), 'CreateWorkspaceCommand.projectRoleName is required');
+        assert(!empty($command->getProjectReadOnlyRoleName()), 'CreateWorkspaceCommand.projectReadOnlyRoleName is required');
+
+        $db = $this->manager->createSession($credentials);
+        try {
+            // root user is also root database
+            $databaseName = $credentials->getPrincipal();
+
+            // TODO jaky stackPrefix?
```

Comment:
Přidaj ho do protobuf commandu tak jak je to jinde.

Reply:
oukej

Reply:
done